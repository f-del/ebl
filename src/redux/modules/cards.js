import update from "immutability-helper";

const CREATE = "CARD/CREATE";
const START = "CARD/START";

const card = title => ({
  Title: title,
  Status: cardStatus.Todo,
  Type: cardType.Task
});

export const cardStatus = {
  Todo: "TODO",
  Inprogress: "INPROGRESS"
};

export const cardType = {
  Task: "TASK"
};

export const createCardSuccess = (id, card) => ({
  type: CREATE,
  payload: { Id: id, ...card }
});

export const createCard = title => {
  return (dispatch, getState, { api }) => {
    const newCard = card(title);
    return api.Tasks.Post(newCard).then(res => {
      console.log("post done,  new Id =" + res.Id);
      dispatch(createCardSuccess(res.Id, newCard));
    });
  };
};

export const startCard = id => {
  if (id === undefined) throw new Error("Id arg is mandatory");
  return {
    type: START,
    payload: {
      Id: id
    }
  };
};

const initialState = {
  list: []
};

export default function(state = initialState, action) {
  console.debug(action);
  switch (action.type) {
    case CREATE:
      return { ...state, list: [...state.list, action.payload] };
    case START:
      let idx = state.list.findIndex(c => c.Id === action.payload.Id);

      var mergeObject = {};
      mergeObject[idx] = { $merge: { Status: cardStatus.Inprogress } };

      return { ...state, list: update(state.list, mergeObject) };

    default:
      return state;
  }
}

export function getAllCards(state) {
  if (state === undefined) throw new Error("State arg is mandatory");
  return state.cards !== undefined ? state.cards.list : [];
}

export function getAllCardsInProgess(state) {
  return getAllCards(state).filter(isCardStatusInProgess);
}
export function getAllCardsTodo(state) {
  return getAllCards(state).filter(isCardStatusToDo);
}

/* Functionnal helper */
function isCardStatusInProgess(card) {
  return card.Status === cardStatus.Inprogress;
}
function isCardStatusToDo(card) {
  return card.Status === cardStatus.Todo;
}
