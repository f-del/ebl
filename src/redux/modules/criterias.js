export const criteriaType = {
  BASIC: "BASIC",
  DEV: "DEV"
};

export function create(id, value) {
  return { Id: id, Value: value };
}

const initialState = {
  list: {
    BASIC: [
      {
        Id: "IdCritBas1",
        Value: false
      }
    ],
    DEV: [
      {
        Id: "IdCritDev1",
        Text: "Development",
        Value: false
      },
      {
        Id: "IdCritDev2",
        Text: "Unit testing",
        Value: false
      }
    ]
  }
};

export default function(state = initialState, action) {
  return state;
}

export function getCriteria(state, type) {
  if (type === undefined) throw new Error("Type argument is mandatory");
  return getState(state).list[type] || [];
}
function getState(state) {
  return state.criterias || {};
}
