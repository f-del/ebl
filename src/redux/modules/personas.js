/***
 *    ██████╗ ███████╗██████╗ ██╗   ██╗ ██████╗███████╗██████╗ ███████╗
 *    ██╔══██╗██╔════╝██╔══██╗██║   ██║██╔════╝██╔════╝██╔══██╗██╔════╝
 *    ██████╔╝█████╗  ██║  ██║██║   ██║██║     █████╗  ██████╔╝███████╗
 *    ██╔══██╗██╔══╝  ██║  ██║██║   ██║██║     ██╔══╝  ██╔══██╗╚════██║
 *    ██║  ██║███████╗██████╔╝╚██████╔╝╚██████╗███████╗██║  ██║███████║
 *    ╚═╝  ╚═╝╚══════╝╚═════╝  ╚═════╝  ╚═════╝╚══════╝╚═╝  ╚═╝╚══════╝
 *
 */

const initialState = {
  list: {
    e6wTJIBsAw5puwwrNEsR: {
      Name: "Feature Team",
      Details: "Meta persona, representing all collaborators working with Paul",
      Needs: [
        "To understand WHAT the Product Owner wants to be able to design collaborativly the HOW",
        "To manage easly the list of things to do to finish the most quickly the User Story"
      ],
      Picture: ""
    },
    uOCF2GLFiJl6fvxdlq4b: {
      Details: "Product Owner for a new product, work in a startup",
      Name: "Paul Deligny",
      Needs: [
        "To have a clear understanding about the product vision & objectives",
        "To write real User Stories, ie have a Customer Driven thinking",
        "To share at the right time and with the rigth level of information the User Stories with his team",
        "To have a clear vision on the job he'll have to do AND the job already or currently done"
      ],
      Picture: ""
    }
  }
};

export default function(state = initialState, action) {
  return state;
}

/***
 *    ███████╗███████╗██╗     ███████╗ ██████╗████████╗ ██████╗ ██████╗ ███████╗
 *    ██╔════╝██╔════╝██║     ██╔════╝██╔════╝╚══██╔══╝██╔═══██╗██╔══██╗██╔════╝
 *    ███████╗█████╗  ██║     █████╗  ██║        ██║   ██║   ██║██████╔╝███████╗
 *    ╚════██║██╔══╝  ██║     ██╔══╝  ██║        ██║   ██║   ██║██╔══██╗╚════██║
 *    ███████║███████╗███████╗███████╗╚██████╗   ██║   ╚██████╔╝██║  ██║███████║
 *    ╚══════╝╚══════╝╚══════╝╚══════╝ ╚═════╝   ╚═╝    ╚═════╝ ╚═╝  ╚═╝╚══════╝
 *
 */
export function getAllPersonas(state) {
  return mapToArrayPersona(getPersonasList(state)) || [];
}

function getPersonasList(state) {
  return getPersonas(validateState(state)).list || {};
}

/* Functionnal helper */
function validateState(state) {
  if (state === undefined) throw new Error("State arg is mandatory");
  return state;
}
function getPersonas(state) {
  return state.personas || {};
}

function mapToArrayPersona(list) {
  return Array.from(Object.keys(list), key => ({
    ...getPersonaId(key),
    ...list[key]
  }));
}
function getPersonaId(key) {
  return {
    Id: key
  };
}
