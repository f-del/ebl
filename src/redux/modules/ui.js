const SELECT_HYPOTHESIS = "UI/SELECT/HYPOTHESIS";
const SELECT_USERSTORY = "UI/SELECT/USER-STORY";

export const selectHypothesis = hypothesis => ({
  type: SELECT_HYPOTHESIS,
  payload: {
    Id: hypothesis.Id
  }
});

export const selectUserStory = userstory => ({
  type: SELECT_USERSTORY,
  payload: {
    Id: userstory.Id
  }
});

const initialState = {
  currentHypothesis: undefined,
  currentUserStory: undefined
};

export default function uiReducer(state = initialState, action) {
  switch (action.type) {
    case SELECT_HYPOTHESIS: {
      // currently Reducer handle Ui logic -> deselect current US on Hypothesis change
      return {
        ...state,
        currentHypothesis: action.payload.Id,
        currentUserStory: undefined
      };
    }
    case SELECT_USERSTORY: {
      return { ...state, currentUserStory: action.payload.Id };
    }

    default:
      return state;
  }
}

export function getHypothesisSelected(state) {
  return state.ui.currentHypothesis;
}
export function getUserStorySelected(state) {
  return state.ui.currentUserStory;
}
