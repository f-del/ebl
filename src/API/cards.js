import { cardStatus } from "../redux/modules/cards";

const mapping = card => {
  card = card._firestore !== undefined ? card.data() : card;
  return {
    ...card,
    Status:
      card.Status === "INPROGRESS" ? cardStatus.INPROGRESS : cardStatus.TODO
  };
};

function post(db) {
  return card => {
    return new Promise((resolve, reject) => {
      db.collection("cards")
        .add(card)
        .then(ref => {
          console.log("Document successfully written!");
          resolve({ id: ref.id });
        })
        .catch(error => {
          console.error("Error writing document: ", error);
          //dispatch(errorSave())
          reject(error);
        });
    });
  };
}

function get(db) {
  return type => {
    return new Promise((resolve, reject) => {
      db.collection("cards")
        .where("Type", "==", type)
        .get()
        .then(function(querySnapshot) {
          let cards = [];

          querySnapshot.forEach(function(doc) {
            let env = mapping(doc);

            cards.push(env);
          });

          resolve(cards);
        });
    });
  };
}

export default db => {
  return { Post: post(db), Get: get(db) };
};
