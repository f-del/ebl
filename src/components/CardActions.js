import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import green from "@material-ui/core/colors/green";
import UiTypography from "@material-ui/core/Typography";
import UiMenuItem from "@material-ui/core/MenuItem";
import UiSelect from "@material-ui/core/Select";
import UiInputLabel from "@material-ui/core/InputLabel";
import UiFormControl from "@material-ui/core/FormControl";
import UiFormControlLabel from "@material-ui/core/FormControlLabel";
import UiFormGroup from "@material-ui/core/FormGroup";
import UiFormHelperText from "@material-ui/core/FormHelperText";
import UiButton from "@material-ui/core/Button";
import UiCheckbox from "@material-ui/core/Checkbox";

import { cardStatus } from "../redux/modules/cards";

const styles = theme => ({
  root: {
    color: green[600],
    "&$checked": {
      color: green[500]
    }
  },
  checked: {},
  button: {
    spacing: theme.spacing.unit
  }
});
function CardActions({ card, onAction, criterias_typology_list, classes }) {
  const displayINPROGRESS = () => (
    <UiFormControl component="fieldset">
      <UiTypography component="legend">Done criterias :</UiTypography>
      <UiFormGroup>
        {card.Criterias.map(crit => (
          <UiFormControlLabel
            key={crit.Id}
            disabled={crit.Value === true}
            control={
              <UiCheckbox
                id={"tdod_" + card.Id + "_" + crit.Id}
                classes={{
                  root: classes.root,
                  checked: classes.checked
                }}
                value={crit.Id}
                checked={crit.Value === true}
                onChange={e => handleCheckBox(e, crit.Id)}
              />
            }
            label={crit.Text}
            labelPlacement="start"
          />
        ))}
      </UiFormGroup>
    </UiFormControl>
  );

  const displayTODO = () => (
    <UiButton
      color="primary"
      size="small"
      className={classes.button}
      onClick={onAction}
    >
      Start
    </UiButton>
  );

  const displayTODO_AffectCriterias = () => (
    <UiFormControl>
      <UiInputLabel htmlFor={`criteriaSelect${card.Id}`}>DoD</UiInputLabel>
      <UiSelect
        id={`criteriaSelect${card.Id}`}
        onChange={handleSelect}
        value=""
      >
        {criterias_typology_list.map((ct, i) => (
          <UiMenuItem key={i} value={Object.keys(ct)[0]}>
            {ct.Text || Object.keys(ct)}
          </UiMenuItem>
        ))}
      </UiSelect>

      <UiFormHelperText>Affect a Definition of Done workflow</UiFormHelperText>
    </UiFormControl>
  );

  if (card.Status === cardStatus.TODO)
    if (card.Criterias !== undefined && card.Criterias.length > 0)
      return displayTODO();
    else return displayTODO_AffectCriterias();

  if (card.Status === cardStatus.INPROGRESS) return displayINPROGRESS();

  return null;
  function handleSelect(e) {
    onAction({ value: e.target.value });
  }

  function handleCheckBox(e, id) {
    onAction({ id: "" + id, value: e.target.checked });
  }
}

CardActions.propTypes = {
  card: PropTypes.object.isRequired,
  onAction: PropTypes.func.isRequired
};

export default withStyles(styles)(CardActions);
