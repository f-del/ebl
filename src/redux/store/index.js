import cards from "../modules/cards";
import { combineReducers } from "redux";

const rootReducer = combineReducers({ cards });
export default rootReducer;
