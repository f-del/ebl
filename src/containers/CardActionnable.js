import PropTypes from "prop-types";
import { connect } from "react-redux";
import {
  cardStatus,
  toggleCardCriteria,
  startCard
} from "../redux/modules/cards";
import CardActions from "../components/CardActions";

const mapStateToProps = (state, ownProps) => ({});

const mapDispatchToProps = (dispatch, ownProps) => ({
  onAction: newValue => {
    if (ownProps.status === cardStatus.TODO) dispatch(startCard(ownProps.id));
    else if (ownProps.status === cardStatus.INPROGRESS)
      dispatch(toggleCardCriteria(ownProps.id, newValue.id, newValue.value));
  }
});

const CardActionnable = connect(
  mapStateToProps,
  mapDispatchToProps
)(CardActions);

CardActionnable.propTypes = {
  id: PropTypes.string.isRequired,
  status: PropTypes.symbol.isRequired
};

export default CardActionnable;
