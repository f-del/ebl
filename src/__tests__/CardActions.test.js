import React from "react";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { createStore, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import reducer from "../redux/store/index";

import UiSelect from "@material-ui/core/Select";
import UiButton from "@material-ui/core/Button";
import UiMenuItem from "@material-ui/core/MenuItem";
import UiCheckbox from "@material-ui/core/Checkbox";
import * as cardsSelector from "../redux/modules/cards";
import { cardStatus } from "../redux/modules/cards";
import CardActions from "../components/CardActions";
import CardActionnable from "../containers/CardActionnable";
import {
  entity_test_created,
  entity_test_created_with_criterias
} from "./datas";
import {
  entity_criteria,
  entity_criteria_type,
  retMockGetCriteria
} from "./redux/criteria.test";
import { criteriaType } from "../redux/modules/criterias";

configure({ adapter: new Adapter() });
let store;

const checkboxActionparams = (id = "1", value = true) => ({ id, value });
const entity_card_created_on_progress = {
  ...entity_test_created_with_criterias,
  Status: cardStatus.INPROGRESS
};
describe("Componant tests", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  test("Initial state in status TODO, without Criterias", () => {
    const mockAction = jest.fn(() => {});
    const wrapper = shallow(
      <CardActions
        card={entity_test_created}
        onAction={mockAction}
        criterias_typology_list={[
          entity_criteria_type(),
          entity_criteria_type("NOTEXT", entity_criteria)
        ]}
      />
    ).dive();

    expect(wrapper.find(UiButton).length).toBe(0);
    expect(wrapper.find(UiSelect).length).toBe(1);
    const listitem = wrapper.find(UiMenuItem);
    expect(listitem.length).toBe(2);

    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });

  test("Expect assert call onAction, on Select a Dod criteria", () => {
    const mockAction = jest.fn(() => {});
    const wrapper = shallow(
      <CardActions
        card={entity_test_created}
        onAction={mockAction}
        criterias_typology_list={[
          entity_criteria_type(),
          entity_criteria_type("NOTEXT", entity_criteria)
        ]}
      />
    ).dive();
    const selectDod = wrapper.find(UiSelect);
    selectDod.simulate("change", { target: { value: "BASIC" } });
    expect(mockAction.mock.calls.length).toBe(1);
    const onActionArg = mockAction.mock.calls[0][0];
    expect(onActionArg).toBeDefined();
    expect(onActionArg).toStrictEqual({ value: "BASIC" });
    expect(wrapper).toMatchSnapshot();
  });

  test("Initial state in status TODO with criteria", () => {
    const mockAction = jest.fn(id => {});
    const wrapper = shallow(
      <CardActions
        card={entity_test_created_with_criterias}
        onAction={mockAction}
      />
    ).dive();

    const btnStart = wrapper.find(UiButton);
    btnStart.simulate("click");
    expect(btnStart.length).toBe(1);
    expect(mockAction.mock.calls.length).toBe(1);
    expect(mockAction.mock.calls[0][0]).not.toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("Initial state in status INPROGRESS", () => {
    const mockAction = jest.fn(() => {});

    entity_card_created_on_progress.Criterias[1].Value = true;
    const wrapper = mount(
      <CardActions
        card={entity_card_created_on_progress}
        onAction={mockAction}
      />
    );

    const checkboxes = wrapper.find(UiCheckbox);
    expect(checkboxes.length).toBeGreaterThan(0);

    const disabled = checkboxes.at(1).props().disabled;
    expect(disabled).toStrictEqual(true);
    expect(checkboxes.at(1).props().checked).toBeTruthy();

    expect(wrapper.find("button").length).toBe(0);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("Click checkbox on Status PROGRESS", () => {
    const mockAction = jest.fn(() => {});
    const wrapper = mount(
      <CardActions
        card={entity_card_created_on_progress}
        onAction={mockAction}
      />
    );

    const checkbox = wrapper.find(UiCheckbox);
    checkbox
      .at(0)
      .props()
      .onChange({ target: { checked: true } });

    expect(mockAction.mock.calls.length).toBe(1);
    expect(mockAction.mock.calls[0][0]).toStrictEqual(checkboxActionparams());
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("Click on all checkbox validate Status", () => {
    const mockAction = jest.fn(() => {});

    const wrapper = mount(
      <CardActions
        card={entity_card_created_on_progress}
        onAction={mockAction}
      />
    );

    const checkbox = wrapper.find(UiCheckbox);
    checkbox.forEach(c => c.props().onChange({ target: { checked: true } }));

    expect(mockAction.mock.calls.length).toBe(2);
    expect(mockAction.mock.calls[0][0]).toStrictEqual(checkboxActionparams());
    expect(mockAction.mock.calls[1][0]).toStrictEqual(
      checkboxActionparams("2")
    );

    entity_card_created_on_progress.Criterias[0].Value = true;
    entity_card_created_on_progress.Criterias[1].Value = true;
    wrapper.setProps({ card: entity_card_created_on_progress });
    const checkboxChecked = wrapper.find(UiCheckbox);
    checkboxChecked.forEach(ckb => {
      expect(ckb.props().disabled).toStrictEqual(true);
      expect(ckb.props().checked).toBeTruthy();
    });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});

describe("Containers tests", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  test("Expect props onAction is defined", () => {
    const wrapper = shallow(
      <CardActionnable
        card={entity_test_created_with_criterias}
        status={cardStatus.TODO}
      />
    );

    expect(wrapper.props("onAction")).toBeDefined();
  });
  test("Call onAction with status TODO and no criteria expect mock on setCriteriasTypology", () => {
    cardsSelector.setCriteriasTypology = jest
      .fn()
      .mockImplementationOnce(id => {
        return dispatch => {
          return id;
        };
      });

    store = createStore(reducer, applyMiddleware(thunk));

    const wrapper = mount(
      <Provider store={store}>
        <CardActionnable card={entity_test_created} />,
      </Provider>
    );

    const selectDod = wrapper.find(UiSelect);
    selectDod
      .at(0)
      .props()
      .onChange({ target: { value: "BASIC" } });
    expect(cardsSelector.setCriteriasTypology.mock.calls.length).toBe(1);
    expect(cardsSelector.setCriteriasTypology.mock.calls[0][0]).toBe("1");
    expect(cardsSelector.setCriteriasTypology.mock.calls[0][1]).toBe(
      criteriaType.BASIC
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("Call onAction with status TODO expect mock on startCard", () => {
    cardsSelector.updateCardStatusForward = jest
      .fn()
      .mockImplementationOnce(id => {
        return dispatch => {
          return id;
        };
      });
    store = createStore(reducer, applyMiddleware(thunk));

    const wrapper = mount(
      <Provider store={store}>
        <CardActionnable card={entity_test_created_with_criterias} />,
      </Provider>
    );
    const btnStart = wrapper.find("button");
    btnStart.simulate("click");
    expect(cardsSelector.updateCardStatusForward.mock.calls.length).toBe(1);
    expect(cardsSelector.updateCardStatusForward.mock.calls[0][0]).toBe("1");
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("Call onAction with status INPROGRESS expect mock on toggleCardCriteria", () => {
    cardsSelector.toggleCardCriteria = jest.fn().mockImplementationOnce(id => {
      return dispatch => {
        return id;
      };
    });
    store = createStore(reducer, applyMiddleware(thunk));

    const wrapper = mount(
      <Provider store={store}>
        <CardActionnable
          card={{
            ...entity_test_created,
            Status: cardStatus.INPROGRESS,
            Criterias: [...retMockGetCriteria]
          }}
        />
        ,
      </Provider>
    );
    const checkbox = wrapper.find('input[type="checkbox"]');
    checkbox.forEach(c => c.simulate("change", { target: { checked: true } }));
    expect(cardsSelector.toggleCardCriteria.mock.calls.length).toBe(1);
    expect(cardsSelector.toggleCardCriteria.mock.calls[0][0]).toBe("1");
    expect(cardsSelector.toggleCardCriteria.mock.calls[0][1]).toBe(
      "IdCritBas1"
    );
    expect(cardsSelector.toggleCardCriteria.mock.calls[0][2]).toBe(true);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
