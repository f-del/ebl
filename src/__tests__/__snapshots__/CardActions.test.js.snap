// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componant tests Click checkbox on Status PROGRESS 1`] = `
<div>
  Done criterias:
  <ul>
    <li
      key="1"
    >
      <input
        checked={false}
        id="tdod_1_1"
        onChange={[Function]}
        readOnly={false}
        type="checkbox"
        value="1"
      />
      <label
        htmlFor="tdod_1_1"
      />
    </li>
    <li
      key="2"
    >
      <input
        checked={true}
        id="tdod_1_2"
        onChange={[Function]}
        readOnly={true}
        type="checkbox"
        value="2"
      />
      <label
        htmlFor="tdod_1_2"
      />
    </li>
  </ul>
</div>
`;

exports[`Componant tests Click on all checkbox validate Status 1`] = `
<CardActions
  card={
    Object {
      "Criterias": Array [
        Object {
          "Id": "1",
          "Value": true,
        },
        Object {
          "Id": "2",
          "Value": true,
        },
      ],
      "Id": "1",
      "Status": Symbol(INPROGRESS),
      "Title": "test",
      "Type": "TASK",
    }
  }
  onAction={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "id": "1",
            "value": true,
          },
        ],
        Array [
          Object {
            "id": "2",
            "value": true,
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <div>
    Done criterias:
    <ul>
      <li
        key="1"
      >
        <input
          checked={true}
          id="tdod_1_1"
          onChange={[Function]}
          readOnly={true}
          type="checkbox"
          value="1"
        />
        <label
          htmlFor="tdod_1_1"
        />
      </li>
      <li
        key="2"
      >
        <input
          checked={true}
          id="tdod_1_2"
          onChange={[Function]}
          readOnly={true}
          type="checkbox"
          value="2"
        />
        <label
          htmlFor="tdod_1_2"
        />
      </li>
    </ul>
  </div>
</CardActions>
`;

exports[`Componant tests Expect assert call onAction, on Select a Dod criteria 1`] = `
<div>
  <select
    onChange={[Function]}
  >
    <option>
      Affect a DoD to task
    </option>
    <option
      key="0"
      value="BASIC"
    >
      BASIC
    </option>
    <option
      key="1"
      value="NOTEXT"
    >
      NOTEXT
    </option>
  </select>
</div>
`;

exports[`Componant tests Initial state in status INPROGRESS 1`] = `
<CardActions
  card={
    Object {
      "Criterias": Array [
        Object {
          "Id": "1",
          "Value": false,
        },
        Object {
          "Id": "2",
          "Value": true,
        },
      ],
      "Id": "1",
      "Status": Symbol(INPROGRESS),
      "Title": "test",
      "Type": "TASK",
    }
  }
  onAction={[MockFunction]}
>
  <div>
    Done criterias:
    <ul>
      <li
        key="1"
      >
        <input
          checked={false}
          id="tdod_1_1"
          onChange={[Function]}
          readOnly={false}
          type="checkbox"
          value="1"
        />
        <label
          htmlFor="tdod_1_1"
        />
      </li>
      <li
        key="2"
      >
        <input
          checked={true}
          id="tdod_1_2"
          onChange={[Function]}
          readOnly={true}
          type="checkbox"
          value="2"
        />
        <label
          htmlFor="tdod_1_2"
        />
      </li>
    </ul>
  </div>
</CardActions>
`;

exports[`Componant tests Initial state in status TODO with criteria 1`] = `
<div>
  <button
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  >
    Start
  </button>
</div>
`;

exports[`Componant tests Initial state in status TODO, without Criterias 1`] = `
<div>
  <select
    onChange={[Function]}
  >
    <option>
      Affect a DoD to task
    </option>
    <option
      key="0"
      value="BASIC"
    >
      BASIC
    </option>
    <option
      key="1"
      value="NOTEXT"
    >
      NOTEXT
    </option>
  </select>
</div>
`;

exports[`Containers tests Call onAction with status INPROGRESS expect mock on toggleCardCriteria 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(CardActions)
    card={
      Object {
        "Criterias": Array [
          Object {
            "Id": "IdCritBas1",
            "Value": "defaultBas",
          },
        ],
        "Id": "1",
        "Status": Symbol(INPROGRESS),
        "Title": "test",
        "Type": "TASK",
      }
    }
  >
    <CardActions
      card={
        Object {
          "Criterias": Array [
            Object {
              "Id": "IdCritBas1",
              "Value": "defaultBas",
            },
          ],
          "Id": "1",
          "Status": Symbol(INPROGRESS),
          "Title": "test",
          "Type": "TASK",
        }
      }
      criterias_typology_list={
        Array [
          Object {
            "BASIC": Array [
              Object {
                "Id": "IdCritBas1",
                "Value": false,
              },
            ],
          },
          Object {
            "DEV": Array [
              Object {
                "Id": "IdCritDev1",
                "Text": "Development",
                "Value": false,
              },
              Object {
                "Id": "IdCritDev2",
                "Text": "Unit testing",
                "Value": false,
              },
            ],
          },
        ]
      }
      onAction={[Function]}
    >
      <div>
        Done criterias:
        <ul>
          <li
            key="IdCritBas1"
          >
            <input
              checked={false}
              id="tdod_1_IdCritBas1"
              onChange={[Function]}
              readOnly={false}
              type="checkbox"
              value="IdCritBas1"
            />
            <label
              htmlFor="tdod_1_IdCritBas1"
            />
          </li>
        </ul>
      </div>
    </CardActions>
  </Connect(CardActions)>
  ,
</Provider>
`;

exports[`Containers tests Call onAction with status TODO and no criteria expect mock on setCriteriasTypology 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(CardActions)
    card={
      Object {
        "Id": "1",
        "Status": Symbol(TODO),
        "Title": "test",
        "Type": "TASK",
      }
    }
  >
    <CardActions
      card={
        Object {
          "Id": "1",
          "Status": Symbol(TODO),
          "Title": "test",
          "Type": "TASK",
        }
      }
      criterias_typology_list={
        Array [
          Object {
            "BASIC": Array [
              Object {
                "Id": "IdCritBas1",
                "Value": false,
              },
            ],
          },
          Object {
            "DEV": Array [
              Object {
                "Id": "IdCritDev1",
                "Text": "Development",
                "Value": false,
              },
              Object {
                "Id": "IdCritDev2",
                "Text": "Unit testing",
                "Value": false,
              },
            ],
          },
        ]
      }
      onAction={[Function]}
    >
      <div>
        <select
          onChange={[Function]}
        >
          <option>
            Affect a DoD to task
          </option>
          <option
            key="0"
            value="BASIC"
          >
            BASIC
          </option>
          <option
            key="1"
            value="DEV"
          >
            DEV
          </option>
        </select>
      </div>
    </CardActions>
  </Connect(CardActions)>
  ,
</Provider>
`;

exports[`Containers tests Call onAction with status TODO expect mock on startCard 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(CardActions)
    card={
      Object {
        "Criterias": Array [
          Object {
            "Id": "1",
            "Value": true,
          },
          Object {
            "Id": "2",
            "Value": true,
          },
        ],
        "Id": "1",
        "Status": Symbol(TODO),
        "Title": "test",
        "Type": "TASK",
      }
    }
  >
    <CardActions
      card={
        Object {
          "Criterias": Array [
            Object {
              "Id": "1",
              "Value": true,
            },
            Object {
              "Id": "2",
              "Value": true,
            },
          ],
          "Id": "1",
          "Status": Symbol(TODO),
          "Title": "test",
          "Type": "TASK",
        }
      }
      criterias_typology_list={
        Array [
          Object {
            "BASIC": Array [
              Object {
                "Id": "IdCritBas1",
                "Value": false,
              },
            ],
          },
          Object {
            "DEV": Array [
              Object {
                "Id": "IdCritDev1",
                "Text": "Development",
                "Value": false,
              },
              Object {
                "Id": "IdCritDev2",
                "Text": "Unit testing",
                "Value": false,
              },
            ],
          },
        ]
      }
      onAction={[Function]}
    >
      <div>
        <button
          onClick={[Function]}
        >
          Start
        </button>
      </div>
    </CardActions>
  </Connect(CardActions)>
  ,
</Provider>
`;
